spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationBuilder.java

```diff
@@ -116,15 +117,20 @@ public ConfigurableApplicationContext run(String[] parentArgs) {
 			apps.add(sinkConfigurer);
 		}
 		List<Class<?>> appsToEmbed = new ArrayList<>();
+		String namespace = null;
 		for (int i = 0; i < apps.size(); i++) {
-			appsToEmbed.add(apps.get(i).getApp());
+			AppConfigurer<?> appConfigurer = apps.get(i);
+			appsToEmbed.add(appConfigurer.getApp());
+			if (namespace == null) {
+				namespace = appConfigurer.namespace;
```

Comment:
So instead of using the first encountered namespace throughout, we should decide for each app whether we want to use the default (generated) value or the value set through the builder (which should be app-specific).